### DISCUSSION TITLE
need add static id for all channels

### DISCUSSION THREAD

**Discussion by TechComet** [iptv-org/community#153] [Ideas]
Hello,

I'm need id for channel
and last update channel info

I'm have channel id in local database

need update channel info if ip-tv update channel info by id

**Comment by BellezaEmporium** [comment]
Please clarify, as your request isn't clear at all. What do you suggest by "adding a static ID for all channels" ?

Isn't the tvg-id already sufficient for that ?

**Reply by TechComet** [reply]
It's not sufficient
multi channels have same `tvg-id`

**Reply by freearhey** [reply]
@TechComet The same channel will have the same `tvg-id`
and if you need the id of each channel link, you can use the url itself as id

**Reply by TechComet** [reply]
@freearhey 

it's doesn't help

I'm download this one time in month programmicly
`https://iptv-org.github.io/iptv/channels.json`

<details>
<summary>Expand</summary>
<br>
  
```
[
  ...
  {
    "name": "CNN",
    "logo": "https://i.imgur.com/ilZJT5s.png",
    "url": "http://ott-cdn.ucom.am/s27/index.m3u8",
    "categories": [
      {
        "name": "News",
        "slug": "news"
      }
    ],
    "countries": [
      {
        "code": "us",
        "name": "United States"
      },
      {
        "code": "ca",
        "name": "Canada"
      }
    ],
    "languages": [
      {
        "code": "eng",
        "name": "English"
      }
    ],
    "tvg": {
      "id": "cnn.us",
      "name": "CNN",
      "url": "http://epg.streamstv.me/epg/guide-usa.xml.gz"
    }
  },
  ...
]
```
</details>

I'm Have `channels` table in database
```
tvg_id
tvg_name
tvg_url
```
I'm add only my channels favitore

I'm need sync any changes data from json file to my database
for that I need static id
if I use tvg_id = multi channels have same id = bad idea
if I use tvg_url = ! = I'm need static id for check if url or data for this static id changed for update my data


I'm sorry my english is not good

**Comment by freearhey** [comment]
In the current version, the API includes a channel ID:

`https://iptv-org.github.io/api/channels.json`

```jsonc
[
  //...
  {
    "id": "CNN.us",
    "name": "CNN",
    //...
  }
  //...
]
```

https://github.com/iptv-org/api?tab=readme-ov-file#channels
